backup.c

```diff
@@ -185,14 +185,19 @@ do_backup_database(parray *backup_list, pgBackupOption bkupopt)
 	/*
 	 * To take incremental backup, the file list of the latest validated
 	 * full database backup is needed.
+	 * There is a problem of increment backup.
```

Comment:
I think the target is only detect something wrong a physical backup file.
so I didn't reflect this part 「or pg_rman's delete command 
with '--force' option」.

Reply:
It's both.

It is a challenging task, 
but it is necessary to check whether the physical backup file is 
corrupted and the backup chain list is regular.
It is difficult to write a detailed example here., however, the explanation is omitted from the backup file 
because it is essential to check that all backup files are normally stored (including physical corruption).

And, I think that the verification(check) of the backup file chain list is 
necessary to ensure that incremental backups and restores operate normally.
For example, if there is a chain list of backup files of 
FB(1gen)->IB(1gen-1)->IB(1gen-2)->IB(1gen-3),
If IB(1gen-2) is forcibly deleted, 
it is fine until IB (1gen-1), but backup and restore will not be possible after that.
If it doesn't detect this, it could get a meaningless backup, and the wrong restore could work.

Therefore, I think @mikecaat  suggestion looks good.

Best regards.
Moon.

Reply:
@huangfumingyue @MoonInsung 
Thanks for your comments!

OK. I understood that there are two problems to be solved.
If so, is it better to separate the issue because (1) and (2) seem to be different problems and need different solutions.

(1) The first is to check files corruption, missing files, and so on only in one backup list.

I couldn't understand this problem clearly because do_validate() already takes charge of checking the file's corruption with CRC.

For example, the case is following.

$ rm ~/tmp/backup/20210527/090151/database/base/12405/1247
> pg_rman restore
(omit)
WARNING: backup file "/home/ikeda/tmp/backup/20210527/090151/database/base/12405/1247" vanished
WARNING: backup "2021-05-27 09:01:51" is corrupted
INFO: restoring database files from the incremental mode backup "2021-05-27 09:01:51"
ERROR: could not open backup file "/home/ikeda/tmp/backup/20210527/090151/database/base/12405/1247": そのようなファイルやディレクトリはありません
(omit)
What am I missing? Is there any corner case?

(2) Second is to check the backup chain is correct. I agree this is not implemented yet.

Reply:
Dear @mikecaat 
Thank you.

Sorry. My explanation was insufficient.

In my opinion,
is check whether the backup file is corrupted is when acquiring a new incremental backup.
In other words, 
(1) Is all backup files of the generation that try incremental backup normal?
(2) Is the backup file of the chain normal?
I thought it was necessary to check.
Of course, this could maybe be the wrong idea because it's my personal opinion.
That's why I think it needs to be discussed later.

Best regards.
Moon.

Reply:
@MoonInsung 
Thanks! I understood. I updated the issue to mention the above.